ironfish/src/syncer.ts

```diff
@@ -442,7 +442,9 @@ export class Syncer {
 
       if (!this.loader) {
         this.logger.info(`Syncing orphan chain from ${peer.displayName}`)
-        this.startSync(peer)
+        if (peer.sequence !== null) {
```

Comment:
We should always have a peer sequence if the peer is identified. This should never happen right? Are we trying to sync from peers that are not identified?

Reply:
Yes, you should always have a peer sequence if the peer is identified. This should never happen. You are also trying to sync from peers that are not identified. Other than fixing the bug like this, there is no other method. Or to fix this bug, you’ll have to to remake hash algorithm.